clang/include/clang/CConv/DeclRewriter.h

```diff
@@ -0,0 +1,69 @@
+//=--DeclRewriter.h-----------------------------------------------*- C++-*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// This file contains the DeclRewriter class which is used to rewrite variable
+// declarations in a program using the checked pointers types solved for by the
+// the conversion tool.
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_CLANG_LIB_CCONV_DECLREWRITER_H
+#define LLVM_CLANG_LIB_CCONV_DECLREWRITER_H
+
+#include "clang/AST/Decl.h"
+#include "clang/AST/Stmt.h"
+#include "clang/AST/ASTContext.h"
+#include "clang/Rewrite/Core/Rewriter.h"
+#include "clang/CConv/RewriteUtils.h"
+
+using namespace llvm;
+using namespace clang;
+
+class DeclRewriter {
+public:
+  DeclRewriter(Rewriter &R, ASTContext &A, GlobalVariableGroups &GP)
+      : R(R), A(A), GP(GP), Skip(DComp(A.getSourceManager())) {}
+
+  // The publicly accessible interface for performing declaration rewritting.
+  // All declaration for variables with checked types in the variable map of
+  // Info are rewritten.
+  static void rewriteDecls(ASTContext &Context, ProgramInfo &Info, Rewriter &R,
+                           std::set<FileID> &TouchedFiles);
+private:
+  Rewriter &R;
+  ASTContext &A;
+  GlobalVariableGroups &GP;
+  // Skip indicates some rewrites that
+  // we should skip because we already applied them, for example, as part
+  // of turning a single line declaration into a multi-line declaration.
+  RSet Skip;
+
+  // TODO: I don't like having this be static.
+  static std::map<std::string, std::string> NewFuncSig;
+
+  // Visit each Decl in toRewrite and apply the appropriate pointer type
```

Comment:
Typo: toRewrite --> ToRewrite

Reply:
And: Discussion of `R`, etc. could go with those variables' definitions, above. What is `S` ?

Reply:
Discussion of variables is stale from before I added RewriteDecl class and pulled some things into fields. I'll update it.